session/common.go

```diff
@@ -30,6 +30,44 @@ type commonData struct {
 
 type contextReader func(session.TokenContext, bool) error
 
+func (x commonData) copyTo(dst *commonData) {
+	if x.idSet {
+		dst.idSet = true
```

Comment:
`x.idSet` may be false while `dst.idSet` is true. See comment in `netmap`: that test should detect such cases

same below